sdk/python/kfp/onprem.py

```diff
@@ -73,3 +87,51 @@ def _use_k8s_secret(task):
         return task
 
     return _use_k8s_secret
+
+
+def add_default_resource_spec(
+    memory_limit: str = None,
+    cpu_limit: str = None,
+    memory_request: str = None,
+    cpu_request: str = None,
+):
+    """Add default resource requests and limits.
+
+    For resource units, refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes.
+
+    Args:
+      memory_limit: optional, memory limit. Format can be 512Mi, 2Gi etc.
+      cpu_limit: optional, cpu limit. Format can be 0.5, 500m etc.
+      memory_request: optional, defaults to memory limit.
+      cpu_request: optional, defaults to cpu limit.
+    """
+    if not memory_request:
+        memory_request = memory_limit
+    if not cpu_request:
+        cpu_limit = cpu_request
+
+    def _add_default_resource_spec(task: dsl.ContainerOp):
+        # Skip tasks which are not container ops.
+        if not isinstance(task, dsl.ContainerOp):
+            return task
+        _apply_default_resource(task, 'cpu', cpu_request, cpu_limit)
+        _apply_default_resource(task, 'memory', memory_request, memory_limit)
```

Comment:
Since I see you return task on line 116, I would guess you want to achieve chained setting?
If that's the case, shouldn't we return task here at the end as well?

Reply:
Missed that!